### ISSUE TITLE
about data_loader

### ISSUE THREAD

**Issue by Jee-King** [uzh-rpg/snn_angular_velocity#5]
Hi, thanks for your great work!

the following codes are lines 29 and 30 of  _data_loader/utils.py_,  the _time_idx_ is in [0,99], does this mean that only one event will be retained if some events with the same coordinates (_ev_xy_) and the same _time_idx_.
```
spike_tensor = torch.zeros((2, self.height, self.width, self.num_time_bins))
spike_tensor[ev_pol.long(), ev_xy[:, 1].long(), ev_xy[:, 0].long(), time_idx.long()] = 1
```
Thank you very much!

**Comment by magehrig** [comment]
Hi @Jee-King 

Yes, this is correct. I think it would be better to instead sum up the number of events per time index instead. Of course, then the SNN pipeline must support this and I don't remember if Slayer does right now.

**Comment by Jee-King** [comment]
Hi, thanks for your reply!

I am reproducing the training process according to the paper. I use the MSE loss function instead of the function described in the paper. But I got a bad performance. Does the MSE loss function have a decisive effect on the results? By the way, could you provide the source code of the loss function in the paper?

**Comment by magehrig** [comment]
I don't think that you will see a big difference if you use a slightly different loss. The challenge of training this SNN lies mostly in making sure that you have gradients for all layers. I used the following loss:

```Python
class RMSELoss:
    def __init__(self):
        self.str_to_idx = {
            'x': 0,
            'y': 1,
            'z': 2,
        }

    def compute(self, input_: torch.Tensor, target: torch.Tensor, time_start_idx: int=None, axis: str=None):
        assert len(input_.shape) == 3
        assert len(target.shape) == 3
        assert input_.shape == target.shape
        assert input_.shape[1] == 3
        """
        :param input: tensor of shape (batch, 3, time)
        :param target: tensor of shape (batch, 3, time)
        :param time_start_idx: Time-index from which to start computing the loss
        :param axis: {'x', 'y', 'z'}
        :return: Root mean squared error
        """
        if axis:
            axis_idx = self.str_to_idx[axis]
            input_ = input_[:, axis_idx, :]
            target = target[:, axis_idx, :]

        if time_start_idx:
            input_ = input_[..., time_start_idx:]
            target = target[..., time_start_idx:]

        return torch.mean(torch.sqrt(torch.mean((input_ - target) ** 2, dim=1)))
```

**Comment by Jee-King** [comment]
Thank you very much for your help!
